distributed/worker.py

```diff
@@ -2855,13 +2889,16 @@ def stimulus_story(
         keys = {e.key if isinstance(e, TaskState) else e for e in keys_or_tasks}
         return [ev for ev in self.stimulus_log if getattr(ev, "key", None) in keys]
 
-    def ensure_communicating(self) -> None:
+    def _ensure_communicating(self, *, stimulus_id: str) -> RecsInstrs:
```

Comment:
Fetching the stimulus_id from outside means that now gather_dep commands will have the stimulus_id of the event that triggered them, e.g.

- compute-task
- acquire-replicas
- find_missing
- GatherDepDoneEvent